systems/controllers/QPCommon.cpp

```diff
@@ -2,29 +2,39 @@
 #include "drakeFloatingPointUtil.h"
 #include "controlUtil.h"
 #include <map>
+#include "lcmUtil.h"
 
 #define LEG_INTEGRATOR_DEACTIVATION_MARGIN 0.05
 
-template<int M, int N>
-void matlabToCArrayOfArrays(const mxArray *source, const int idx, const char *fieldname, double *destination)  {
-  // Matlab arrays come in as column-major data. To represent a matrix in C++, as we do in our LCM messages, we need an array of arrays. But that convention results in a row-major storage, so we have to be careful about how we copy data in. 
-  const mxArray *field = mxGetField(source, idx, fieldname);
-  if (!field) {
-    field = mxGetPropertySafe(source, idx, fieldname);
-  }
-  sizecheck(field, M, N);
-  Map<Matrix<double, M, N>>A(mxGetPrSafe(field));
-  // C is row-major, matlab is column-major
-  Matrix<double, N, M> A_t = A.transpose();
-  memcpy(destination, A_t.data(), sizeof(double)*M*N);
-  return;
-}
-
 double logisticSigmoid(double L, double k, double x, double x0) {
   // Compute the value of the logistic sigmoid f(x) = L / (1 + exp(-k(x - x0)))
   return L / (1.0 + exp(-k * (x - x0)));
 }
 
+PiecewisePolynomial<double> matlabToPiecewisePolynomial(const mxArray* pobj, int index) {
```

Comment:
This function has some hardcoded dimensions in it, so not this one. But the ones in constructQPLocomotionPlanmex will be moved to drakeMexUtil in a subsequent pull request (once drakeMexUtil exists). The code that calls this function should then be changed to call the future drakeMexUtil version.